data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/admin/ReadinessTest.java

```diff
@@ -13,113 +13,136 @@
+    handler.handleRequest(request);
+
+    assertThat(request.getResponseCode()).isEqualTo(SC_OK);
+    assertThat(request.getResponseBody()).isNull();
   }
 
   @Test
   public void shouldReturnUnavailableWhenStoreNotAvailable() throws Exception {
-    final Readiness handler =
-        new Readiness(syncDataProvider, chainDataProvider, network, jsonProvider);
+    request.setOptionalQueryParameter(TARGET_PEER_COUNT, "1234");
     when(chainDataProvider.isStoreAvailable()).thenReturn(false);
 
-    handler.handle(context);
-    verifyCacheStatus(CACHE_NONE);
-    verifyStatusCode(SC_SERVICE_UNAVAILABLE);
+    handler.handleRequest(request);
+
+    assertThat(request.getResponseCode()).isEqualTo(SC_SERVICE_UNAVAILABLE);
+    assertThat(request.getResponseBody()).isNull();
   }
 
   @Test
   public void shouldReturnUnavailableWhenStartingUp() throws Exception {
-    final Readiness handler =
-        new Readiness(syncDataProvider, chainDataProvider, network, jsonProvider);
+    request.setOptionalQueryParameter(TARGET_PEER_COUNT, "1234");
     when(chainDataProvider.isStoreAvailable()).thenReturn(true);
     when(syncService.getCurrentSyncState()).thenReturn(SyncState.START_UP);
 
-    handler.handle(context);
-    verifyCacheStatus(CACHE_NONE);
-    verifyStatusCode(SC_SERVICE_UNAVAILABLE);
+    handler.handleRequest(request);
+
+    assertThat(request.getResponseCode()).isEqualTo(SC_SERVICE_UNAVAILABLE);
+    assertThat(request.getResponseBody()).isNull();
   }
 
   @Test
   public void shouldReturnUnavailableWhenSyncing() throws Exception {
-    final Readiness handler =
-        new Readiness(syncDataProvider, chainDataProvider, network, jsonProvider);
+    request.setOptionalQueryParameter(TARGET_PEER_COUNT, "1234");
     when(chainDataProvider.isStoreAvailable()).thenReturn(true);
     when(syncService.getCurrentSyncState()).thenReturn(SyncState.SYNCING);
 
-    handler.handle(context);
-    verifyCacheStatus(CACHE_NONE);
-    verifyStatusCode(SC_SERVICE_UNAVAILABLE);
+    handler.handleRequest(request);
+
+    assertThat(request.getResponseCode()).isEqualTo(SC_SERVICE_UNAVAILABLE);
+    assertThat(request.getResponseBody()).isNull();
   }
 
   @Test
-  public void shouldReturnBadRequestWhenWrongTargetPeerCountParam() throws Exception {
-    final Readiness handler =
-        new Readiness(syncDataProvider, chainDataProvider, network, jsonProvider);
+  public void shouldReturnBadRequestWhenWrongTargetPeerCountParam() {
+    request.setOptionalQueryParameter(TARGET_PEER_COUNT, "a");
     when(chainDataProvider.isStoreAvailable()).thenReturn(true);
     when(syncService.getCurrentSyncState()).thenReturn(SyncState.IN_SYNC);
-    when(context.queryParamMap()).thenReturn(Map.of(TARGET_PEER_COUNT, List.of("a")));
 
-    handler.handle(context);
-    verifyCacheStatus(CACHE_NONE);
-    verifyStatusCode(SC_BAD_REQUEST);
+    assertThatThrownBy(() -> handler.handleRequest(request))
+        .isInstanceOf(IllegalArgumentException.class);
   }
```

Comment:
this test actually demonstrates that code question I raised, so i'm happy with that...